classes/submissionFile/maps/Schema.inc.php

```diff
@@ -112,6 +133,27 @@ protected function mapByProperties(array $props, SubmissionFile $item): array
                 continue;
             }
 
+            if ($prop === 'genreIsPrimary') {
```

Comment:
I've always felt that this should be an enum [primary, supplementary, dependent] instead of two settings. I don't think it's sensible for a file genre to be `isDependent` and `isSupplementary` at the same time, although the configuration allows that. The way that files are actually handled in the article landing page and galley views assumes they are one or the other. Thoughts?

For now I'll adjust the API to return a separate property for the genre which includes `dependent` and `supplementary` info.

Reply:
:+1: That's sensible, I do wish we'd gone with an enum rather than two settings. We could revisit that in the future but of course we'd have to consider what happens for users who have already designated files as both.